src/component/ResponsiveContainer.tsx

```diff
@@ -60,28 +67,31 @@ export const ResponsiveContainer = forwardRef(
         containerWidth: containerRef.current.clientWidth,
         containerHeight: containerRef.current.clientHeight,
       };
-    };
-
-    const updateDimensionsImmediate = () => {
-      if (!mounted) {
-        return;
-      }
+    }, []);
 
+    const updateDimensionsImmediate = useCallback(() => {
       const newSize = getContainerSize();
 
       if (newSize) {
-        const { containerWidth: oldWidth, containerHeight: oldHeight } = sizes;
         const { containerWidth, containerHeight } = newSize;
 
-        if (containerWidth !== oldWidth || containerHeight !== oldHeight) {
-          setSizes({ containerWidth, containerHeight });
-        }
+        setSizes(currentSizes => {
```

Comment:
this has to be a bug fix since referencing previous values with the state itself is an antipattern and causes weird renders. Nice!